src/metax_api/models/catalog_record_v2.py

```diff
@@ -99,6 +99,7 @@ def _pre_create_operations(self):
 
         self.research_dataset['metadata_version_identifier'] = generate_uuid_identifier()
         self.identifier = generate_uuid_identifier()
+        self._handle_issued_date()
```

Comment:
This works for datasets that are published right away, but not on datasets that are first drafts and then published. There's also third way of publishing datasets with merge_draft. It should be investigated how that works as well and it would be good to write appropriate tests for these scenarios as well. Note that these only apply for v2 API

Reply:
Related to this https://jira.eduuni.fi/browse/CSCFAIRMETA-642, issued date is not generated for drafts and it is tested that they won't have this field.